src/rules/no_implicit_declare_namespace_export.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
+use super::{Context, LintRule};
+use crate::handler::{Handler, Traverse};
+use crate::{Program, ProgramRef};
+use deno_ast::{view as ast_view, SourceRanged};
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct NoImplicitDeclareNamespaceExport;
+
+const CODE: &str = "no-implicit-declare-namespace-export";
+const MESSAGE: &str =
+  "Implicit exports in ambient namespaces are discouraged to
+use";
+const HINT: &str = "Try adding an `export {};` to the top of the namespace to disable this behavior";
+
+impl LintRule for NoImplicitDeclareNamespaceExport {
+  fn new() -> Arc<Self> {
+    Arc::new(NoImplicitDeclareNamespaceExport)
+  }
+
+  fn tags(&self) -> &'static [&'static str] {
+    &[]
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn lint_program(&self, _context: &mut Context, _program: ProgramRef<'_>) {
+    unreachable!();
+  }
+
+  fn lint_program_with_ast_view(
+    &self,
+    context: &mut Context,
+    program: Program<'_>,
+  ) {
+    if context.file_name().ends_with(".d.ts") {
```

Comment:
This is different from what is written in the documentation. The document said this lint rule works for both `ts` and `d.ts` files.

Reply:
Thanks! It is now fixed.